src/simulation.c

```diff
@@ -33,17 +34,19 @@ void free_simulation_history(SimulationHistory *simulation_historyp) {
   free(simulation_historyp);
 }
 
-void insert_reaction(SimulationHistory *simulation_historyp, int reaction) {
+void insert(SimulationHistory *simulation_historyp, int reaction, double time) {
```

Comment:
Since C doesn't have namespaces, it isn't a good idea to use identifiers like `insert`. Maybe `insert_history_element` since it is longer than `insert_reaction`